PWGPP/EMCAL/AliAnalysisTaskEMCALAlig.cxx

```diff
@@ -122,126 +69,9 @@ void AliAnalysisTaskEMCALAlig::UserCreateOutputObjects()
-        fOutput->Add(fPositronYRes[i]);
-        
-        //All tracks
-        
-        fAllMatchedTracksPhiRes[i] = new TH2F(Form("fAllMatchedTracksPhiRes%d",i),"Distance (phi) of EMCAL cluster in all matched tracks ;#Delta#phi;#Delta#eta",100,0,2*TMath::Pi(),100,-0.3,0.3);
-        fOutput->Add(fAllMatchedTracksPhiRes[i]);
-        
-        fAllMatchedTracksEtaRes[i] = new TH2F(Form("fAllMatchedTracksEtaRes%d",i),"Distance (phi) of EMCAL cluster in all matched tracks ;#Delta#phi;#Delta#eta",100,0,2*TMath::Pi(),100,-0.3,0.3);
-        fOutput->Add(fAllMatchedTracksEtaRes[i]);
-        
-        fElectronPhiResRU[i] = new TH2F(Form("fElectronPhiResRU%d",i),"Distance (phi) of EMCAL cluster in ID electrons RU;#phi;#Delta#phi",100,0,2*TMath::Pi(),1000,-0.3,0.3);
-        fOutput->Add(fElectronPhiResRU[i]);
-        
-        fElectronEtaResRU[i] = new TH2F(Form("fElectronEtaResRU%d",i),"Distance (eta) of EMCAL clusterin ID electrons RU;#phi;#Delta#eta",100,0,2*TMath::Pi(),1000,-0.3,0.3);
-        fOutput->Add(fElectronEtaResRU[i]);
-        
-        fElectronXResRU[i] = new TH2F(Form("fElectronXResRU%d",i),"Distance (x) of EMCAL cluster in ID electrons RU;#phi;#Delta x",100,0,2*TMath::Pi(),1000,-20,20);
-        fOutput->Add(fElectronXResRU[i]);
-        
-        fElectronYResRU[i] = new TH2F(Form("fElectronYResRU%d",i),"Distance (y) of EMCAL clusterin ID electrons RU;#phi;#Delta y",100,0,2*TMath::Pi(),1000,-20,20);
-        fOutput->Add(fElectronYResRU[i]);
-        
-        fElectronPositronZResRU[i] = new TH2F(Form("fElectronPositronZRes%dRU",i),"Distance (z) of EMCAL clusterin ID electrons and positrons RU;#phi;#Delta z",100,0,2*TMath::Pi(),1000,-20,20);
-        fOutput->Add(fElectronPositronZResRU[i]);
-        
-        //Positrons
-        
-        fPositronPhiResRU[i] = new TH2F(Form("fPositronPhiResRU%d",i),"Distance (phi) of EMCAL cluster in ID positrons RU;#Delta#phi;#Delta#eta",100,0,2*TMath::Pi(),1000,-0.3,0.3);
-        fOutput->Add(fPositronPhiResRU[i]);
-        
-        fPositronEtaResRU[i] = new TH2F(Form("fPositronEtaResRU%d",i),"Distance (eta) of EMCAL clusterin ID positrons RU;#phi;#Delta#eta",100,0,2*TMath::Pi(),1000,-0.3,0.3);
-        fOutput->Add(fPositronEtaResRU[i]);
-        
-        fPositronXResRU[i] = new TH2F(Form("fPositronXResRU%d",i),"Distance (x) of EMCAL cluster in ID positrons RU;#phi;#Delta x",100,0,2*TMath::Pi(),1000,-20,20);
-        fOutput->Add(fPositronXResRU[i]);
-        
-        fPositronYResRU[i] = new TH2F(Form("fPositronYResRU%d",i),"Distance (y) of EMCAL clusterin ID positrons RU;#phi;#Delta y",100,0,2*TMath::Pi(),1000,-20,20);
-        fOutput->Add(fPositronYResRU[i]);
-        
-        
-    }
-    
-
-    
+    ElectronTree = new TTree("electron_information","electron_information");
+    ElectronTree->Branch("electrons", &ElectronInformation);
+    fOutput->Add(ElectronTree);
```

Comment:
I'm not an expert here, but I was under the impression that it was preferred to have a separate output slot for trees. Perhaps someone more knowledgeable can comment?

Reply:
Yes, definitely better to use a separate slot, rather than adding it to a TList. This way you can use a TChain to combine and analyse the result from multiple trains (e.g. if you want to analyse results from different datasets=trains combined together). It may also have some improvements in the performance, but not 100% sure that's the case.